test/upgrade/suite_execution.go

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2020 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package upgrade
+
+import (
+	"testing"
+
+	"go.uber.org/zap"
+)
+
+type suiteExecution struct {
+	suite         *Suite
+	configuration Configuration
+	failed        bool
+	logger        *zap.SugaredLogger
+	stopSignals   []StopSignal
```

Comment:
You need to hold channel in both places: sender and receiver. Background Operation is holding it as received at `BackgroundContext.Stop`. This place is required to send stop messages when we like to stop those running tests.

Reply:
Okay. This might be good to do. But, I will not move it to background operation. That type is general, so that implementor can pass his own type if he like to. I will move this to a wrapper of BackgroundOperation.